### PR TITLE
[FLINK-16056][runtime][tests] fix CFRO creation in tests

### PR DESCRIPTION
**Pull request by rkhachatryan**
## What is the purpose of the change

*ContinuousFileReaderOperator was changed recently to be created using factory. But some tests create it using constructor directly. This causes problems with deserialization.*

## Brief change log

  - *fix test to actually fail on exception - previously error was only logged*
  - *use factory in tests*
  - *remove testing constructor*
  - *make class package-private*

## Verifying this change

This change is already covered by existing tests: `ContinuousFileProcessingITCase`, `ContinuousFileProcessingMigrationTest`, `ContinuousFileProcessingTest`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-fs-tests/src/test/java/org/apache/flink/hdfstests/ContinuousFileProcessingITCase.java [hunk 1]
```diff
@@ -140,29 +141,25 @@ public void testProgram() throws Exception {
...
+		CompletableFuture<Void> jobFuture = new CompletableFuture<>();
+		new Thread(() -> {
+			try {
+				env.execute("ContinuousFileProcessingITCase Job.");
+				jobFuture.complete(null);
+			} catch (Exception e) {
+				Throwable th = e;
+				for (int depth = 0; depth < 20; depth++) {
+					if (th instanceof SuccessException) {
+						return;
```
Comment (AHeise) [inline flink-fs-tests/src/test/java/org/apache/flink/hdfstests/ContinuousFileProcessingITCase.java]: should that also complete future?

**Comment by rkhachatryan** [comment]
Thanks for feedback @AHeise, 
I've addressed the issue.

**Comment by rkhachatryan** [comment]
> LGTM % could you expand the `CFRO` abbreviation in the commit message?

Done.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java [hunk 2]
```diff
@@ -70,7 +70,6 @@
 import org.apache.flink.streaming.api.datastream.DataStreamSource;
 import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
 import org.apache.flink.streaming.api.functions.source.ContinuousFileMonitoringFunction;
-import org.apache.flink.streaming.api.functions.source.ContinuousFileReaderOperator;
```
Comment (pnowojski) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java]: One more thing. Doesn't removal of line brake the java docs for this class (`@Public` so quite important)? 

Replace `{@link ContinuousFileReaderOperator readers}` with full class reference?

Comment (rkhachatryan) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java]: Yes, but the actual operator type is now hidden in the factory, so I'd rather remove the reference at all and leave just plain "readers".

Comment (pnowojski) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java]: I think a reference to the code might be helpful for some more power users. I would be also fine with referencing `ContinuousFileReaderOperatorFactory` itself, as that wouldn't explode the comment's length.

Comment (rkhachatryan) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java]: IMO javadoc of public API shouldn't talk about the implementation classes at all, unless there is a need (here there isn't). And "power users" should be able to go inside the method and see the implementation:)

Comment (pnowojski) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java]: Ok, in that case please update the java docs.

Comment (rkhachatryan) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java]: Done.

**Comment by pnowojski** [comment]
Merged. Can you close the jira ticket @rkhachatryan ?

**Comment by rkhachatryan** [comment]
Done. Thanks.

**Pull request merged**